lib/improver/utilities/temporal.py

```diff
@@ -63,10 +133,9 @@ def find_required_lead_times(cube):
     else:
         if cube.coords("time") and cube.coords("forecast_reference_time"):
             try:
-                cube.coord("time").convert_units(
-                    "hours since 1970-01-01 00:00:00")
+                cube.coord("time").convert_units(time_units)
                 cube.coord("forecast_reference_time").convert_units(
-                    "hours since 1970-01-01 00:00:00")
+                    forecast_reference_time_units)
             except ValueError as err:
                 msg = "For time/forecast_reference_time: {}".format(err)
                 raise ValueError(msg)
```

Comment:
Would it be constraining a user too much to add a check here in case any of the calculated forecast periods are negative? If I get my data wrong when forcing the FRT I could end up with negative forecast periods and would want to know about it. Not sure how Ver would cope.

Reply:
I've added an exception if negative forecast periods are generated. There may be reasons why this could just be a warning, however, for the moment, I've made this an exception.

Reply:
:white_check_mark: